File: 백준/Silver/11728. 배열 합치기/배열 합치기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        StringBuilder sb = new StringBuilder();

        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        int[] A = new int[N];
        int[] B = new int[M];
        
        st = new StringTokenizer(br.readLine());
        for(int i = 0; i < N; i++) {
            A[i] = Integer.parseInt(st.nextToken());
        }

        st = new StringTokenizer(br.readLine());
        for(int i = 0; i < M; i++) {
            B[i] = Integer.parseInt(st.nextToken());
        }
        
        int Astart = 0, Bstart = 0;

        while(Astart < N || Bstart < M) {
            int num = 0;
            if(Astart == N) {
                num = B[Bstart++];
            } else if(Bstart == M) {
                num = A[Astart++];
            } else if(A[Astart] < B[Bstart]) {
                num = A[Astart++];
            } else {
                num = B[Bstart++];
            }
            
            sb.append(num).append(" ");
        }

        System.out.println(sb.toString());

    }
}

File: 백준/Silver/11728. 배열 합치기/배열 합치기.kt
import java.util.StringTokenizer

//2 2
//3 5
//2 9
fun main() {
    var st = StringTokenizer(readln())
    val a = st.nextToken().toInt()
    val b = st.nextToken().toInt()

    val Aarr = IntArray(a) { 0 }
    val Barr = IntArray(b) { 0 }

    st = StringTokenizer(readln())
    for(i in 0..a-1) {
        Aarr[i] = st.nextToken().toInt()
    }

    st = StringTokenizer(readln())
    for(i in 0..b-1) {
        Barr[i] = st.nextToken().toInt()
    }

    var Aidx = 0
    var Bidx = 0
    val max = a + b

    val sb = StringBuilder()
    while (Aidx + Bidx < max) {
        if(Aidx >= a) {
            sb.append(Barr[Bidx++]).append(" ")
        } else if(Bidx >= b) {
            sb.append(Aarr[Aidx++]).append(" ")
        } else if(Aarr[Aidx] > Barr[Bidx]) {
            sb.append(Barr[Bidx++]).append(" ")
        } else {
            sb.append(Aarr[Aidx++]).append(" ")
        }
    }

    print(sb)
}